crypto/pem/pem_test.cc

```diff
@@ -43,3 +58,141 @@ TEST(PEMTest, NoRC4) {
+
+    // Write the ASN1_INTEGER to a PEM-formatted string
+    ASSERT_TRUE(PEM_ASN1_write(i2d_ASN1_INTEGER_void, "ASN1 INTEGER",
+                               pem_file.get(), asn1_int.get(), nullptr, nullptr,
+                               0, nullptr, nullptr));
+
+    rewind(pem_file.get());
+    // Read the ASN1_INTEGER back from the PEM-formatted string
+    bssl::UniquePtr<ASN1_INTEGER> read_integer((ASN1_INTEGER *)PEM_ASN1_read(
+        d2i_ASN1_INTEGER_void, "ASN1 INTEGER", pem_file.get(), nullptr,
+        nullptr, nullptr));
+    ASSERT_TRUE(read_integer);
+
+    // Check if the read ASN1_INTEGER has the same value as the original
+    long read_value = ASN1_INTEGER_get(read_integer.get());
+    ASSERT_EQ(original_value, read_value);
+  }
+}
+
+const char* kPemRsaPrivateKey = "-----BEGIN ENCRYPTED PRIVATE KEY-----\n"
+    "MIHsMFcGCSqGSIb3DQEFDTBKMCkGCSqGSIb3DQEFDDAcBAhz3vU103jx3wICCAAw\n"
+    "DAYIKoZIhvcNAgkFADAdBglghkgBZQMEASoEEA6vMhRLgHZuHFa+eiecYCgEgZDB\n"
+    "E8EOzjGQuu4D0TVAjOa3Peb9/MzQz3t09m5pvNBFKrEl96gefpZdni5qQk34ukj9\n"
+    "/kryXymP72m4Ch7vbmew08x5x9L69BpfsQLF1yyvAJVtEZ0a1Zqcn5veuoH2WtJT\n"
+    "ZTrZtc5Eb+tAjMVzRXPZ80cUwCbbpb9KHUX8spwtG10I1VxJ18X31FVpGORdr0A=\n"
+    "-----END ENCRYPTED PRIVATE KEY-----";
+
+
+TEST(PEMTest, ReadPrivateKeyPem) {
+  bssl::UniquePtr<BIO> read_bio(BIO_new_mem_buf(kPemRsaPrivateKey, strlen(kPemRsaPrivateKey)) );
+  ASSERT_TRUE(read_bio.get());
+  bssl::UniquePtr<EC_KEY> ec_key(PEM_read_bio_ECPrivateKey(read_bio.get(), nullptr, pem_password_callback, (void*)SECRET));
+  ASSERT_TRUE(ec_key);
+  const EC_GROUP* p256 = EC_GROUP_new_by_curve_name(NID_X9_62_prime256v1);
+  ASSERT_EQ(p256, EC_KEY_get0_group(ec_key.get()));
+}
+
+
+TEST(PEMTest, WriteReadRSAPem) {
+  bssl::UniquePtr<RSA> rsa(RSA_new());
+  ASSERT_TRUE(rsa);
+  bssl::UniquePtr<BIGNUM> bn(BN_new());
+  ASSERT_TRUE(bn);
+  BN_set_u64(bn.get(), RSA_F4);
+#ifdef BORINGSSL_FIPS
+  ASSERT_TRUE(RSA_generate_key_fips(rsa.get(), 2048, nullptr));
+#else
+  ASSERT_TRUE(RSA_generate_key_ex(rsa.get(), 2048, bn.get(), nullptr));
+#endif
+
+  bssl::UniquePtr<BIO> write_bio(BIO_new(BIO_s_mem()));
+  ASSERT_TRUE(write_bio.get());
+  const EVP_CIPHER* cipher = EVP_get_cipherbynid(NID_aes_256_cbc);
+  ASSERT_TRUE(cipher);
+  ASSERT_TRUE(PEM_write_bio_RSAPrivateKey(write_bio.get(), rsa.get(), cipher, (unsigned char*)SECRET, (int)strlen(SECRET), nullptr, nullptr));
+
+  const uint8_t* content;
+  size_t content_len;
+  BIO_mem_contents(write_bio.get(), &content, &content_len);
+
+  bssl::UniquePtr<BIO> read_bio(BIO_new_mem_buf(content, content_len) );
+  ASSERT_TRUE(read_bio.get());
+  bssl::UniquePtr<RSA> rsa_read(PEM_read_bio_RSAPrivateKey(read_bio.get(), nullptr, pem_password_callback, (void*)SECRET));
+  ASSERT_TRUE(rsa_read);
+  ASSERT_EQ(0, BN_cmp(RSA_get0_n(rsa.get()), RSA_get0_n(rsa_read.get())));
+}
+
+TEST(PEMTest, WriteReadECPem) {
+  bssl::UniquePtr<EC_KEY> ec_key(EC_KEY_new());
+  ASSERT_TRUE(ec_key.get());
```

Comment:
Minor nit, but should we be consistent with how we do UniquePtr assertions here and in other places
```suggestion
  ASSERT_TRUE(ec_key);
```